integrations/oci/oci-secrets-mp-config-source/src/main/java/io/helidon/integrations/oci/secrets/mp/configsource/AdpSuppliers.java

```diff
@@ -0,0 +1,271 @@
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import com.oracle.bmc.Region;
+import com.oracle.bmc.auth.BasicAuthenticationDetailsProvider;
+import com.oracle.bmc.auth.ConfigFileAuthenticationDetailsProvider;
+import com.oracle.bmc.auth.InstancePrincipalsAuthenticationDetailsProvider;
+import com.oracle.bmc.auth.InstancePrincipalsAuthenticationDetailsProvider.InstancePrincipalsAuthenticationDetailsProviderBuilder;
+import com.oracle.bmc.auth.ResourcePrincipalAuthenticationDetailsProvider;
+import com.oracle.bmc.auth.ResourcePrincipalAuthenticationDetailsProvider.ResourcePrincipalAuthenticationDetailsProviderBuilder;
+import com.oracle.bmc.auth.SimpleAuthenticationDetailsProvider;
+import com.oracle.bmc.auth.SimpleAuthenticationDetailsProvider.SimpleAuthenticationDetailsProviderBuilder;
+import com.oracle.bmc.auth.SimplePrivateKeySupplier;
+import com.oracle.bmc.auth.StringPrivateKeySupplier;
+
+import static java.lang.System.Logger;
+
+/**
+ * A utility class containing methods that produce {@link Supplier}s of {@link BasicAuthenticationDetailsProvider}
+ * instances of various kinds.
+ */
+final class AdpSuppliers {
+
+    private static final Pattern COMMA_SPLITTER = Pattern.compile("(?<!\\\\),"); // non-escaped comma
+
+    private static final String DEFAULT_OCI_AUTH_PRIVATE_KEY_PATH =
+        Paths.get(System.getProperty("user.home"), ".oci", "oci_api_key.pem").toString();
+
+    private static final String DEFAULT_OCI_CONFIG_PROFILE = "DEFAULT";
+
+    private static final int DEFAULT_OCI_IMDS_TIMEOUT_MILLIS = 100;
+
+    private static final Logger LOGGER = System.getLogger(AdpSuppliers.class.getName());
+
+    private AdpSuppliers() {
+        super();
+    }
+
+    @SuppressWarnings("checkstyle:linelength")
+    static Optional<Supplier<SimpleAuthenticationDetailsProvider>> simple(Function<? super String, ? extends Optional<String>> c) {
+        return simple(c, SimpleAuthenticationDetailsProvider::builder);
+    }
+
+    @SuppressWarnings("checkstyle:linelength")
+    static Optional<Supplier<SimpleAuthenticationDetailsProvider>> simple(Function<? super String, ? extends Optional<String>> c,
+                                                                          Supplier<? extends SimpleAuthenticationDetailsProviderBuilder> bs) {
+        return simple(c, bs, b -> b::build);
+    }
+
+    @SuppressWarnings("checkstyle:linelength")
+    static Optional<Supplier<SimpleAuthenticationDetailsProvider>> simple(Function<? super String, ? extends Optional<String>> c,
+                                                                          Supplier<? extends SimpleAuthenticationDetailsProviderBuilder> bs,
+                                                                          Function<? super SimpleAuthenticationDetailsProviderBuilder, ? extends Supplier<SimpleAuthenticationDetailsProvider>> f) {
```

Comment:
This will use `OciExtension` from the runtime module?

I guess its too early to review this PR.  Stopping here and will review when it comes out of draft.